snapcraft/internal/meta/_snap_packaging.py

```diff
@@ -272,6 +279,16 @@ def _update_yaml_with_defaults(config_data, schema):
                     )
                 )
 
+    app_schema = schema["apps"]["patternProperties"]["^[a-zA-Z0-9](?:-?[a-zA-Z0-9])*$"][
+        "properties"
+    ]
+    # Set default adapter
+    for app in config_data.get("apps", {}).values():
+        if "adapter" not in app:
+            with contextlib.suppress(KeyError):
```

Comment:
if you suppress this you aren't really setting the default adapter, so why is this scenario at play here?

Reply:
Good point, I'll remove it.

Reply:
Fixed.